desktop/core/src/desktop/js/utils/hooks/UseFileUpload/useChunkUpload.ts

```diff
@@ -0,0 +1,151 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useEffect, useState } from 'react';
+import useSaveData from '../useSaveData';
+import useQueueProcessor from '../useQueueProcessor';
+import {
+  DEFAULT_CHUNK_SIZE,
+  DEFAULT_CONCURRENT_CHUNK_UPLOAD,
+  FileUploadStatus
+} from '../../constants/storageBrowser';
+import useLoadData from '../useLoadData';
+import { TaskServerResponse, TaskStatus } from '../../../reactComponents/TaskBrowser/TaskBrowser';
+import {
+  createChunks,
+  getChunksCompletePayload,
+  getChunkItemPayload,
+  getChunkSinglePayload,
+  getStatusHashMap,
+  getTotalChunk,
+  UploadChunkItem,
+  UploadItem
+} from './util';
+
+interface UseUploadQueueResponse {
+  addFiles: (item: UploadItem[]) => void;
+  removeFile: (item: UploadItem) => void;
+  isLoading: boolean;
+}
+
+interface ChunkUploadOptions {
+  onStatusUpdate: (item: UploadItem, newStatus: FileUploadStatus) => void;
+  onComplete: () => void;
+}
+
+const useChunkUpload = ({
+  onStatusUpdate,
+  onComplete
+}: ChunkUploadOptions): UseUploadQueueResponse => {
+  const [pendingItems, setPendingItems] = useState<UploadItem[]>([]);
+
+  const { save } = useSaveData(undefined, {
+    postOptions: {
+      qsEncodeData: false,
+      headers: {
+        'Content-Type': 'multipart/form-data'
+      }
+    }
+  });
+
+  const updateItemStatus = (serverResponse: TaskServerResponse[]) => {
+    const statusMap = getStatusHashMap(serverResponse);
+
+    const remainingItems = pendingItems.filter(item => {
+      const status = statusMap[item.uuid];
+      if (status === TaskStatus.Success || status === TaskStatus.Failure) {
+        const ItemStatus = TaskStatus.Success ? FileUploadStatus.Uploaded : FileUploadStatus.Failed;
+        onStatusUpdate(item, ItemStatus);
+        return false;
+      }
+      return true;
+    });
+    if (remainingItems.length === 0) {
+      onComplete();
+    }
+    setPendingItems(remainingItems);
+  };
+
+  const { data: tasksStatus } = useLoadData<TaskServerResponse[]>(
+    '/desktop/api2/taskserver/get_taskserver_tasks/',
```

Comment:
How do you know that the get_taskserver_tasks are for the tasks we are interested in? Does it use the session id to only show my tasks? but what if I have other tasks running that are not file uploading?

Reply:
Great question. I am not very sure about the internal working of this API.
@athithyaaselvam Can you please help us on this question.